// angularx-flatpickr/esm2022/lib/flatpickr.module.mjs
import { NgModule, InjectionToken, } from '@angular/core';
import { FlatpickrDirective } from './flatpickr.directive';
import { FlatpickrDefaults, } from './flatpickr-defaults.service';
import * as i0 from "@angular/core";
export const USER_DEFAULTS = new InjectionToken('flatpickr defaults');
export function defaultsFactory(userDefaults) {
    const defaults = new FlatpickrDefaults();
    Object.assign(defaults, userDefaults);
    return defaults;
}
export class FlatpickrModule {
    static forRoot(userDefaults = {}) {
        return {
            ngModule: FlatpickrModule,
            providers: [
                {
                    provide: USER_DEFAULTS,
                    useValue: userDefaults,
                },
                {
                    provide: FlatpickrDefaults,
                    useFactory: defaultsFactory,
                    deps: [USER_DEFAULTS],
                },
            ],
        };
    }
    static { this.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: FlatpickrModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule }); }
    static { this.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "14.0.0", version: "17.3.12", ngImport: i0, type: FlatpickrModule, declarations: [FlatpickrDirective], exports: [FlatpickrDirective] }); }
    static { this.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: FlatpickrModule }); }
}
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "17.3.12", ngImport: i0, type: FlatpickrModule, decorators: [{
            type: NgModule,
            args: [{
                    declarations: [FlatpickrDirective],
                    exports: [FlatpickrDirective],
                }]
        }] });
